rules/coding-style/src/Rector/String_/UseClassKeywordForClassNameResolutionRector.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\String_;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Expr\BinaryOp\Concat;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Scalar\String_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\String_\UseClassKeywordForClassNameResolutionRector\UseClassKeywordForClassNameResolutionTest
+ */
+final class UseClassKeywordForClassNameResolutionRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Use `class` keyword for class name resolution in string instead of hardcoded string reference',
+            [
+                new CodeSample(
+                    <<<'PHP'
+$this->write(
+                '<?php if (App\SomeClass::createCache($netteCacheStorage, %var,  $this->global->cacheStack, %node.array?)) { ?>',
+                md5('md5')
+            );
+PHP
+                    ,
+                    <<<'PHP'
+$this->write(
+                '<?php if (' . \App\SomeClass::class . '::createCache($netteCacheStorage, %var,  $this->global->cacheStack, %node.array?)) { ?>',
+                \md5('md5')
+            );
+PHP
```

Comment:
It's better to have code samples as short as possible, as people often look through all 500 rules and need to scan quickly the differences.

E.g.


```diff
-$value = 'App\SomeClass::someMethod()';
+$value = \App\SomeClass::class . '::someMethod()';
```

It also makes writing test easier.

Reply:
Ready